### PR TITLE
You got mail!

### PR DESCRIPTION
**Pull request by SK-Fast**
hai, i add comand n updates n fix n u h stuff!!

+ lookup command
+ guild command
+ stud2brick now had warning
+ uncase sensitive commands
+ URL parsing to ID for lookup/game/user

### CODE REVIEW

**Comment by bluefield-creator** [comment]
The bot should not take URL arguments at all, makes it a bit pointless having ids. Also bloats up the chat if we were to use it he bot in that scenario, my personal recommendation is to keep it as id only.

I'll write a full analysis later

**Comment by bluefield-creator** [comment]
Also commands can be done not case sensitive by modifying no the command string on index.ts, exports.ts should and is required to use Pascal naming for the variables.

**Comment by bluefield-creator** [comment]
Change "stud warning" to just informational alert, it's not really anything "risky " and the conversion is fictional

**Comment by bluefield-creator** [comment]
Some of the command functions you sent don't have a returned type of Promise<Message<Boolean>> if i remember the type

**Comment by bluefield-creator** [comment]
URL parsing parses anything with numbers.

Not just "id" query, try using some of the built-in libraries for parsing.

image

**Comment by bluefield-creator** [comment]
And make the url parser as a function on utils.

**Comment by SK-Fast** [comment]
kkkkkkkkkkkkkkkkkkkkkkkk

File: src/utils/stringUtils.ts [hunk 1]
```diff
@@ -15,7 +15,51 @@ class stringUtils {
...
+            spitted: spitted_url
+        }
+    }
+
+    /**
+     * ### Get Numbers From URL
+     * @param string URL that want to be parsed
+     * @returns array Numbers that has been found in URL
+    */
+    public static async GetNumbersFromURL(string: string) {
```
Comment (bluefield-creator) [inline src/utils/stringUtils.ts]: Improper naming, use camel case!

File: src/utils/stringUtils.ts [hunk 2]
```diff
@@ -15,7 +15,51 @@ class stringUtils {
...
+
+    public static async URLParse(string: string) {
+        let url = new URL(string);
+        let searchParams_url = url.searchParams
+        let spitted_url:Array<any> = string.split("/")
+
+        return {
+            URLData: url,
+            searchParams: searchParams_url,
+            spitted: spitted_url
```
Comment (bluefield-creator) [inline src/utils/stringUtils.ts]: splitted*

File: src/utils/stringUtils.ts (hunk 1 above)
Comment (bluefield-creator) [reply src/utils/stringUtils.ts]: Rename argument "string" to "url"

File: src/utils/stringUtils.ts [hunk 3]
```diff
@@ -15,7 +15,51 @@ class stringUtils {
...
+        }
+    }
+
+    /**
+     * ### Get Numbers From URL
+     * @param string URL that want to be parsed
```
Comment (bluefield-creator) [inline src/utils/stringUtils.ts]: Use proper JSDoc formatting.
* @param `{ typeHere }` `nameOfTheVariable` `Description of the variable`
```js
/**
* @param {string} url URL to be inserted.
*

File: src/utils/stringUtils.ts [hunk 4]
```diff
@@ -15,7 +15,51 @@ class stringUtils {
...
+     * @returns array Numbers that has been found in URL
+    */
+    public static async GetNumbersFromURL(string: string) {
+        let result = await stringUtils.URLParse(string)
```
Comment (bluefield-creator) [inline src/utils/stringUtils.ts]: Use const here.

File: src/utils/stringUtils.ts [hunk 5]
```diff
@@ -15,7 +15,51 @@ class stringUtils {
...
+        let splitted = result.spitted
+
+        let returnresult:Array<any> = []
```
Comment (bluefield-creator) [inline src/utils/stringUtils.ts]: camel case and Type should be:

```js
const returnResult : number[] = [];

```

File: src/utils/stringUtils.ts [hunk 6]
```diff
@@ -15,7 +15,51 @@ class stringUtils {
...
+
+        splitted.forEach(function (item:string) {
```
Comment (bluefield-creator) [inline src/utils/stringUtils.ts]: Use ES6 Syntax, this is typescript not JavaScript.

```js
splitted.forEach(item => {})
```

`number` type is inherited from the array, meaning it should always be a number.

Comment (bluefield-creator) [reply src/utils/stringUtils.ts]: Thus no need for types on `item` which is not a string?

File: src/utils/stringUtils.ts [hunk 7]
```diff
@@ -15,7 +15,51 @@ class stringUtils {
...
+     * @param string URL that want to be parsed
+     * @returns Dictionary
+     * {
+     *  URLData: origin URL result,
+        searchParams: Search parms, (To get parms, use .get("key") method)
+        spitted: Splitted URL (['https://','','polytoria.com'])
+     * }
+    */
+
+    public static async URLParse(string: string) {
```
Comment (bluefield-creator) [inline src/utils/stringUtils.ts]: Change `string` argument to `url`

File: src/utils/stringUtils.ts [hunk 8]
```diff
@@ -15,7 +15,51 @@ class stringUtils {
...
+        let url = new URL(string);
+        let searchParams_url = url.searchParams
```
Comment (bluefield-creator) [inline src/utils/stringUtils.ts]: Dont mix snake case with camel case with pascal case... use camelCase only.

searchParamsUrl

File: src/utils/stringUtils.ts [hunk 9]
```diff
@@ -15,7 +15,51 @@ class stringUtils {
...
+        let spitted_url:Array<any> = string.split("/")
```
Comment (bluefield-creator) [inline src/utils/stringUtils.ts]: Should be a `const` and the type should be `string[]` with a space after the variable.

Variable name must be camelCase. Not snake case or pascal case.
```ts
const splittedUrl : string[] = url.split("/")
```

File: src/utils/stringUtils.ts [hunk 10]
```diff
@@ -15,7 +15,51 @@ class stringUtils {
...
      */
-    public static async customReplace(string: string, keyword: string, replace: string) {
+     public static async customReplace(string: string, keyword: string, replace: string) {
         return string.replace(new RegExp(keyword, "g"), replace);
     }
+
+    /**
+     * ### URL Parse
+     * @param string URL that want to be parsed
+     * @returns Dictionary
```
Comment (bluefield-creator) [inline src/utils/stringUtils.ts]: "Dictionary" does not exist in JavaScript, in javascript they are objects.
Use proper JSDoc formatting.

File: src/utils/stringUtils.ts [hunk 11]
```diff
@@ -15,7 +15,51 @@ class stringUtils {
...
+     * {
```
Comment (bluefield-creator) [inline src/utils/stringUtils.ts]: Use proper example, this is not what it returns.

File: src/utils/stringUtils.ts [hunk 12]
```diff
@@ -15,7 +15,51 @@ class stringUtils {
...
+    */
+    public static async GetNumbersFromURL(string: string) {
+        let result = await stringUtils.URLParse(string)
+        let splitted = result.spitted
+
+        let returnresult:Array<any> = []
+
+        splitted.forEach(function (item:string) {
+            if (isNaN(parseInt(item)) == false) {
+                returnresult.push(item)
```
Comment (bluefield-creator) [inline src/utils/stringUtils.ts]: camelCase

File: src/utils/stringUtils.ts [hunk 13]
```diff
@@ -15,7 +15,51 @@ class stringUtils {
...
+            }
+        })
+
+        return returnresult
```
Comment (bluefield-creator) [inline src/utils/stringUtils.ts]: camelCase

**Review by bluefield-creator — CHANGES_REQUESTED** [review]
Fix all of these before merguing. Dont work on anything else.

File: package.json [hunk 14]
```diff
@@ -9,7 +9,8 @@
 	},
 	"scripts": {
 		"build": "npx tsc",
-		"start": "node dist/index.js"
+		"start": "node dist/index.js",
+		"buildstart": "npx tsc && node dist/index.js"
```
Comment (bluefield-creator) [inline package.json]: Remove this, build process is not buildstart.

File: src/commands/guild.ts [hunk 15]
```diff
@@ -0,0 +1,44 @@
+import fetch from 'node-fetch';
+import { Message, MessageEmbed, MessageSelectMenu } from 'discord.js'
+import { userUtils } from '../utils/userUtils.js'
+export async function guild(message: Message, _arguments: string[]) {
```
Comment (bluefield-creator) [inline src/commands/guild.ts]: No return type.

File: src/exports.ts [hunk 16]
```diff
@@ -11,7 +12,8 @@ const commands = {
     studToBrick: currencyCommands.studToBrick,
     s2b:         currencyCommands.studToBrick,
     game:        game,
-    cookie:      cookie
+    cookie:      cookie,
+    guild:      guild
```
Comment (bluefield-creator) [inline src/exports.ts]: Align this

**Review by bluefield-creator — CHANGES_REQUESTED** [review]
Change these too

File: src/commands/user/lookup.ts [hunk 17]
```diff
@@ -1,13 +1,66 @@
...
-): void | Promise<Message<boolean>> {
-	if (!_arguments)
-		return message.channel.send('We require the username argument!');
-	
-	
-}
+import fetch from 'node-fetch';
+import { Message, MessageEmbed, MessageSelectMenu } from 'discord.js'
+
+var URLPattern = new RegExp('^(https?:\\/\\/)?'+
```
Comment (bluefield-creator) [inline src/commands/user/lookup.ts]: DONT USE VAR

We use typescript for a reason. Thus use `let` and `const`.

Also what the fuck is this regex

File: src/commands/user/lookup.ts [hunk 18]
```diff
@@ -1,13 +1,66 @@
...
+    '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|'+
+    '((\\d{1,3}\\.){3}\\d{1,3}))'+
+    '(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*'+
+    '(\\?[;&a-z\\d%_.~+=-]*)?'+
+    '(\\#[-a-z\\d_]*)?$','i');
+
+export async function lookUp(message: Message, _arguments: string[]) {
+
+	let apiURL = `https://api.polytoria.com/v1/games/info?id`;
```
Comment (bluefield-creator) [inline src/commands/user/lookup.ts]: _arguments[1] should be:
```
username
```
or
```
id
```
Use switch cases to select the API to use, do not use URL.

File: src/commands/user/lookup.ts [hunk 19]
```diff
@@ -1,13 +1,66 @@
...
+import { Message, MessageEmbed, MessageSelectMenu } from 'discord.js'
+
+var URLPattern = new RegExp('^(https?:\\/\\/)?'+
+    '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|'+
+    '((\\d{1,3}\\.){3}\\d{1,3}))'+
+    '(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*'+
+    '(\\?[;&a-z\\d%_.~+=-]*)?'+
+    '(\\#[-a-z\\d_]*)?$','i');
+
+export async function lookUp(message: Message, _arguments: string[]) {
```
Comment (bluefield-creator) [inline src/commands/user/lookup.ts]: Return type.

File: src/commands/user/lookup.ts (hunk 17 above)
Comment (bluefield-creator) [reply src/commands/user/lookup.ts]: Also pretty sure there's url builders builtin js.

**Review by bluefield-creator — CHANGES_REQUESTED** [review]

**Comment by skooids** [comment]
Get rekt noob

**Comment by Kqpa** [comment]
Dam

**Comment by Alyxsqrd** [comment]
RIP BOZO 😱

File: src/commands/user/lookup.ts (hunk 17 above)
Comment (bluefield-creator) [reply src/commands/user/lookup.ts]: ^^

**Comment by StarManTheGamer** [comment]
ELIMINATED

**Comment by SK-Fast** [comment]
korbo

File: src/commands/user/lookup.ts [hunk 20]
```diff
@@ -1,20 +1,20 @@
...
-			apiURL = "https://api.polytoria.com/v1/users/user?id=" + _arguments[0].replace(/\D/g,'') // Cut string out, we need only ID.
-		} else { // If it's not URL, It would be username
-			apiURL = "https://api.polytoria.com/v1/users/getbyusername?username=" + _arguments[0]
-
-		}
-	} else { // If it's numberic, It would be ID
-		apiURL = "https://api.polytoria.com/v1/users/user?id=" + _arguments[0]
+	switch (_arguments[1]) {
+		case "user":
+			apiURL = `https://api.polytoria.com/v1/users/getbyusername?username=${_arguments[0]}`;
```
Comment (bluefield-creator) [inline src/commands/user/lookup.ts]: Formatting, why is there a space here?

Comment (bluefield-creator) [reply src/commands/user/lookup.ts]: Also why are you doing
```js
let apiURL = ...
```
if its not actually being used at all?

File: package.json [hunk 21]
```diff
@@ -15,13 +15,6 @@
 		"dotenv": "^10.0.0",
 		"node-fetch": "^3.1.0"
 	},
-	"scripts": {
```
Comment (bluefield-creator) [inline package.json]: Did you just remove scripts and types? Why?

File: src/commands/guild.ts [hunk 22]
```diff
@@ -1,12 +1,17 @@
 import fetch from 'node-fetch';
 import { Message, MessageEmbed, MessageSelectMenu } from 'discord.js'
 import { userUtils } from '../utils/userUtils.js'
+import { apiErrorHandler } from '../utils/apiErrorHandler.js'
+
 export async function guild(message: Message, _arguments: string[]) {
```
Comment (bluefield-creator) [inline src/commands/guild.ts]: Still no return type, change apiErrorHandler to responseHandler

File: src/commands/guild.ts [hunk 23]
```diff
@@ -1,12 +1,17 @@
...
 	const parsed = _arguments[0].replace(/[^0-9]/g,'')
 	const apiURL = `https://api.polytoria.com/v1/guild/info?id=${parsed}`;
 
 	const response = await fetch(apiURL);
 	const data: any = await response.json();
+	const errresult = apiErrorHandler.CheckError(response,data)
```
Comment (bluefield-creator) [inline src/commands/guild.ts]: Use camel case.

File: src/commands/guild.ts [hunk 24]
```diff
@@ -1,12 +1,17 @@
...
+	if (errresult.HasError == true) return message.channel.send(`${errresult.DisplayText}`)
```
Comment (bluefield-creator) [inline src/commands/guild.ts]: CAMEL CASE

File: src/commands/user/lookup.ts [hunk 25]
```diff
@@ -19,7 +20,10 @@ export async function lookUp(message: Message, _arguments: string[]) {
 
 	const response = await fetch(apiURL);
 	const data: any = await response.json();
-	if (data.Success !== true) return message.channel.send('There was an unexpected error.')
+	const errresult = apiErrorHandler.CheckError(response,data)
```
Comment (bluefield-creator) [inline src/commands/user/lookup.ts]: Camel case...

File: src/exports.ts [hunk 26]
```diff
@@ -1,5 +1,5 @@
-import * as userCommands from './commands/user.js'
-import {ping} from './commands/ping.js'
+import * as userCommands from './commands/user.js';
+import { ping } from './commands/ping.js';
```
Comment (bluefield-creator) [inline src/exports.ts]: Why did you add this? It's against code style guidelines, check eslint.

File: src/utils/apiErrorHandler.ts [hunk 27]
```diff
@@ -0,0 +1,49 @@
+let DisplayTexts: any = {
```
Comment (bluefield-creator) [inline src/utils/apiErrorHandler.ts]: Should be a const, and under an Error Type.

File: src/utils/apiErrorHandler.ts [hunk 28]
```diff
@@ -0,0 +1,49 @@
...
+     * @param data Gets request.json from node-fetch
+     * @returns Dictionary
+     * {
+            HasError: false,
+            StatusCode: 0,
+            DisplayText: "Error message visibles to user",
+            ActualError: "Actual Error from server"
+        }
+     */
+    public static CheckError(response: any,data: any) {
```
Comment (bluefield-creator) [inline src/utils/apiErrorHandler.ts]: Camel case

File: src/utils/apiErrorHandler.ts [hunk 29]
```diff
@@ -0,0 +1,49 @@
...
+    "Invalid user ID.": "I don't see the player with that id, maybe try again.. or If you're searching using username, Try type \" username\" after your targetted user s' username!"
+}
+
+export class apiErrorHandler {
+    /**
+     * Check Error
+     * @param response Gets request type from node-fetch
+     * @param data Gets request.json from node-fetch
+     * @returns Dictionary
+     * {
```
Comment (bluefield-creator) [inline src/utils/apiErrorHandler.ts]: camel case

File: src/utils/apiErrorHandler.ts [hunk 30]
```diff
@@ -0,0 +1,49 @@
...
+    "Invalid username.": "I don't see the player with that username, maybe try again.. or If you're searching using ID, Try type \" id\" after your targetted user id!",
+    "Invalid user ID.": "I don't see the player with that id, maybe try again.. or If you're searching using username, Try type \" username\" after your targetted user s' username!"
+}
+
+export class apiErrorHandler {
+    /**
+     * Check Error
+     * @param response Gets request type from node-fetch
+     * @param data Gets request.json from node-fetch
+     * @returns Dictionary
```
Comment (bluefield-creator) [inline src/utils/apiErrorHandler.ts]: Dictionary does not exist in javascript, use proper naming.

Comment (bluefield-creator) [reply src/utils/apiErrorHandler.ts]: Most of your mistakes were notified in the past 4 reviews, you still commit them. We have eslint, and prettier for a reason, use the tools.

File: src/utils/apiErrorHandler.ts [hunk 31]
```diff
@@ -0,0 +1,49 @@
...
+     * {
+            HasError: false,
+            StatusCode: 0,
+            DisplayText: "Error message visibles to user",
+            ActualError: "Actual Error from server"
+        }
+     */
+    public static CheckError(response: any,data: any) {
+
+        let result: any = {
```
Comment (bluefield-creator) [inline src/utils/apiErrorHandler.ts]: Add result type, should be a const too.

File: src/utils/stringUtils.ts [hunk 32]
```diff
@@ -1,83 +1,83 @@
...
+     * @returns Replaced String.
+     */
+     public static async customReplace(string: string, keyword: string, replace: string) {
+        return string.replace(new RegExp(keyword, "g"), replace);
+    }
+
+    /**
+     * ### URL Parse
+     * @param string URL that want to be parsed
+     * @returns Dictionary
```
Comment (bluefield-creator) [inline src/utils/stringUtils.ts]: Dictionary does not exist in javascript.

File: src/utils/stringUtils.ts [hunk 33]
```diff
@@ -1,83 +1,83 @@
...
+     * {
+     *  URLData: origin URL result,
```
Comment (bluefield-creator) [inline src/utils/stringUtils.ts]: Use proper example parameter.

File: src/utils/stringUtils.ts [hunk 34]
```diff
@@ -1,83 +1,83 @@
...
+     * }
+    */
+
+    public static URLParse(string: string) {
+        let url = new URL(string);
+        let searchParams_url = url.searchParams
+        let spitted_url:Array<any> = string.split("/")
+
+        return {
+            URLData: url,
```
Comment (bluefield-creator) [inline src/utils/stringUtils.ts]: Camel case.

File: src/utils/stringUtils.ts [hunk 35]
```diff
@@ -1,83 +1,83 @@
...
+            spitted: spitted_url
+        }
+    }
+
+    /**
+     * ### Get Numbers From URL
+     * @param string URL that want to be parsed
+     * @returns array Numbers that has been found in URL
+    */
+    public static GetNumbersFromURL(string: string) {
```
Comment (bluefield-creator) [inline src/utils/stringUtils.ts]: Camel case.

Comment (bluefield-creator) [reply src/utils/stringUtils.ts]: And I told you, string had to be named to url for less confusions.

File: src/utils/stringUtils.ts [hunk 36]
```diff
@@ -1,83 +1,83 @@
...
+
+        let returnresult:Array<any> = []
+
+        splitted.forEach(function (item:string) {
+            if (isNaN(parseInt(item)) == false) {
+                returnresult.push(item)
+            }
+        })
+
+        return returnresult
```
Comment (bluefield-creator) [inline src/utils/stringUtils.ts]: Camel case.

File: src/utils/stringUtils.ts [hunk 37]
```diff
@@ -1,83 +1,83 @@
...
+     * @param string URL that want to be parsed
+     * @returns array Numbers that has been found in URL
+    */
+    public static GetNumbersFromURL(string: string) {
+        let result = stringUtils.URLParse(string)
+        let splitted = result.spitted
+
+        let returnresult:Array<any> = []
+
+        splitted.forEach(function (item:string) {
```
Comment (bluefield-creator) [inline src/utils/stringUtils.ts]: Space on types.

File: src/utils/stringUtils.ts [hunk 38]
```diff
@@ -1,83 +1,83 @@
...
+    /**
+     * ### Get Numbers From URL
+     * @param string URL that want to be parsed
+     * @returns array Numbers that has been found in URL
+    */
+    public static GetNumbersFromURL(string: string) {
+        let result = stringUtils.URLParse(string)
+        let splitted = result.spitted
+
+        let returnresult:Array<any> = []
```
Comment (bluefield-creator) [inline src/utils/stringUtils.ts]: Use `any[]` not `Array<any>` 

Also camel case again.

File: src/utils/stringUtils.ts [hunk 39]
```diff
@@ -1,83 +1,83 @@
...
+        return returnresult
+
+    }
+
+    /**
+     * ### Is URL
+     * @param string string that you want to checks that's url or not
+     * @returns bool returns if it were url or not
+     */
+    public static IsURL(string: string) {
```
Comment (bluefield-creator) [inline src/utils/stringUtils.ts]: Camel case again.

File: src/utils/stringUtils.ts [hunk 40]
```diff
@@ -1,83 +1,83 @@
...
+        let test_url;
+
+        // Try stuff, If URL fail to parse, that means it's not vaild URL.
+        try {
+            test_url = new URL(string)
```
Comment (bluefield-creator) [inline src/utils/stringUtils.ts]: camel case.

File: src/utils/stringUtils.ts [hunk 41]
```diff
@@ -1,83 +1,83 @@
...
+
+    }
+
+    /**
+     * ### Is URL
+     * @param string string that you want to checks that's url or not
+     * @returns bool returns if it were url or not
+     */
+    public static IsURL(string: string) {
+        let test_url;
```
Comment (bluefield-creator) [inline src/utils/stringUtils.ts]: inappropiate naming

File: src/utils/stringUtils.ts [hunk 42]
```diff
@@ -1,83 +1,83 @@
...
+
+        // Try stuff, If URL fail to parse, that means it's not vaild URL.
+        try {
+            test_url = new URL(string)
+        } catch(_) {
+            return false
```
Comment (bluefield-creator) [inline src/utils/stringUtils.ts]: Incorrect use of try and catch functionality.

File: src/utils/stringUtils.ts [hunk 43]
```diff
@@ -1,83 +1,83 @@
...
+        }
+
+        return test_url.protocol === "http:" || test_url.protocol === "https:"; // Allow only site protocols. filter out something like javascript:void(0)
```
Comment (bluefield-creator) [inline src/utils/stringUtils.ts]: Use this properly please...

**Comment by SK-Fast** [comment]
wooo camel, ok

**Comment by bluefield-creator** [comment]
> wooo camel, ok

Do you know what camel case is?

**Comment by SK-Fast** [comment]
> > wooo camel, ok
> 
> Do you know what camel case is?

isCaramelCase

File: src/utils/stringUtils.ts [hunk 44]
```diff
@@ -75,7 +75,7 @@ export class stringUtils {
      * @param string string that you want to checks that's url or not
      * @returns bool returns if it were url or not
      */
-    public static StrisURL(string: string) {
+    public static strIsURL(string: string) {
```
Comment (bluefield-creator) [inline src/utils/stringUtils.ts]: Rename just to isURL, and once more and again... parameter name has to be url...

File: src/utils/stringUtils.ts [hunk 45]
```diff
@@ -75,7 +75,7 @@ export class stringUtils {
...
         return !!url_pattern.test(string);
```
Comment (bluefield-creator) [inline src/utils/stringUtils.ts]: Camel case means ALL of the code has to be camel case.

File: src/utils/stringUtils.ts [hunk 46]
```diff
@@ -1,3 +1,10 @@
+var url_pattern = new RegExp('^(https?:\\/\\/)?'+
```
Comment (bluefield-creator) [inline src/utils/stringUtils.ts]: Why did you bring this back? also `var` is not typescript.

File: src/utils/stringUtils.ts [hunk 47]
```diff
@@ -30,7 +37,7 @@ export class stringUtils {
      * }
     */
 
-    public static URLParse(string: string) {
+    public static urlParse(string: string) {
```
Comment (bluefield-creator) [inline src/utils/stringUtils.ts]: You missed it again?? CHANGE `string` to `url` my god. Parameter names need to be concise.

File: src/utils/stringUtils.ts [hunk 48]
```diff
@@ -30,7 +37,7 @@ export class stringUtils {
...
         let url = new URL(string);
         let searchParams_url = url.searchParams
```
Comment (bluefield-creator) [inline src/utils/stringUtils.ts]: What the fuck? 

DONT MIX camelCase with snake_case!

Don't use Array<any> use any[];

Add a space between types!

File: src/commands/currency/studToBrick.ts [hunk 49]
```diff
@@ -3,16 +3,22 @@ import {Message, MessageEmbed} from 'discord.js'
...
-		fields: [
-			{name: 'studs', value: _arguments[0], inline: true},
-			{name: 'bricks', value: `${studs}`, inline: true}
-		],
-		description: `**${_arguments[0]} <:stud:905987085347983411> ↔️ ${studs} <:brick:905987077995376640>**`
-	})
+    studs /= 15
+    studs = Math.round(studs)
+    let studremoved = Number(_arguments[0]) % 15
+    let studshould = studs - studremoved
```
Comment (bluefield-creator) [inline src/commands/currency/studToBrick.ts]: camel case

File: src/commands/game.ts [hunk 50]
```diff
@@ -1,11 +1,18 @@
...
 export async function game(message: Message, _arguments: string[]) {
-	const apiURL = `https://api.polytoria.com/v1/games/info?id=${_arguments[0]}`
+	const parsed = _arguments[0].replace(/[^0-9]/g,'')
+	const apiURL = `https://api.polytoria.com/v1/games/info?id=${parsed}`;
 
-	const response = await fetch(apiURL)
-	const data: any = await response.json()
+	const response = await fetch(apiURL);
+	const data: any = await response.json();
+	const errresult = responseHandler.checkError(response,data)
```
Comment (bluefield-creator) [inline src/commands/game.ts]: camel case

File: src/utils/stringUtils.ts [hunk 51]
```diff
@@ -54,7 +54,7 @@ export class stringUtils {
      * @param string URL that want to be parsed
      * @returns array Numbers that has been found in URL
     */
-    public static GetNumbersFromURL(string: string) {
+    public static getNumbersFromURL(string: string) {
```
Comment (bluefield-creator) [inline src/utils/stringUtils.ts]: change `string` parameter to `url`

File: src/utils/stringUtils.ts [hunk 52]
```diff
@@ -54,7 +54,7 @@ export class stringUtils {
...
         let result = stringUtils.urlParse(string)
```
Comment (bluefield-creator) [inline src/utils/stringUtils.ts]: Should be a const.

File: src/utils/stringUtils.ts [hunk 53]
```diff
@@ -54,7 +54,7 @@ export class stringUtils {
...
         let splitted = result.spitted
```
Comment (bluefield-creator) [inline src/utils/stringUtils.ts]: should be a const, and array type.

File: src/utils/stringUtils.ts [hunk 54]
```diff
@@ -1,4 +1,4 @@
-var url_pattern = new RegExp('^(https?:\\/\\/)?'+
+const url_pattern = new RegExp('^(https?:\\/\\/)?'+
```
Comment (bluefield-creator) [inline src/utils/stringUtils.ts]: Why is this here? It's unused in favor of builtin javascript URl parsing.

File: src/utils/responseHandler.ts [hunk 55]
```diff
@@ -18,7 +18,7 @@ export class responseHandler {
      */
     public static checkError(response: any,data: any) {
 
-        let result: any = {
+        const result: any = {
```
Comment (bluefield-creator) [inline src/utils/responseHandler.ts]: infer `result` type. If you need it, write it down. Should be relatively easy.

```ts

type result {
   value: dataType
}
```

File: src/utils/responseHandler.ts [hunk 56]
```diff
@@ -18,7 +18,7 @@ export class responseHandler {
...
             HasError: false,
             StatusCode: 0,
             DisplayText: "Unknown Error.",
```
Comment (bluefield-creator) [inline src/utils/responseHandler.ts]: camel case...

File: src/commands/game.ts [hunk 57]
```diff
@@ -1,11 +1,18 @@
...
+	if (errresult.HasError == true) return message.channel.send(`${errresult.DisplayText}`)
```
Comment (bluefield-creator) [inline src/commands/game.ts]: camelcase

File: src/commands/user/lookup.ts [hunk 58]
```diff
@@ -1,7 +1,72 @@
-import {Message} from 'discord.js'
+import fetch from 'node-fetch';
+import { Message, MessageEmbed, MessageSelectMenu } from 'discord.js'
+import { responseHandler } from '../../utils/responseHandler.js'
+
+export async function lookUp(message: Message, _arguments: string[]) {
+
+	let apiURL;
```
Comment (bluefield-creator) [inline src/commands/user/lookup.ts]: Infer type `IapiEndpoints`

File: src/utils/responseHandler.ts [hunk 59]
```diff
@@ -0,0 +1,51 @@
...
+     * @param data Gets request.json from node-fetch
+     * @returns
+     * {
+            HasError: boolean,
+            StatusCode: number,
+            DisplayText: string,
+            ActualError: string
+        }
+     */
+    public static checkError(response: Response,data: any) {
```
Comment (bluefield-creator) [inline src/utils/responseHandler.ts]: space between comas

File: src/utils/stringUtils.ts [hunk 60]
```diff
@@ -1,21 +1,83 @@
...
+     *  URLData: string,
+        searchParams: URLSearchParams,
+        spitted: Array<string>
+     * }
+    */
+
+    public static urlParse(url: string) {
+        let urlConstruct = new URL(url);
+        let searchParams_url = urlConstruct.searchParams
+        let spitted_url: Array<string> = url.split("/")
```
Comment (bluefield-creator) [inline src/utils/stringUtils.ts]: Use type `string[]` not `Array<string>`

**Review by bluefield-creator — CHANGES_REQUESTED** [review]

**Comment by bluefield-creator** [comment]
I'm reviewing only one more batch of updates. No more. Make sure you fix everything, otherwise not getting mergued.

File: src/commands/user/lookup.ts (hunk 58 above)
Comment (SK-Fast) [reply src/commands/user/lookup.ts]: wdym `IapiEndpoints`  ?     ? ?

Comment (bluefield-creator) [reply src/commands/user/lookup.ts]: Create a type that holds API endpoints...

```ts

type IapiEndpoints = "url1" | "url2" | "url3"...
```

Comment (SK-Fast) [reply src/commands/user/lookup.ts]: > Create a type that holds API endpoints...
> 
> ```ts
> type IapiEndpoints = "url1" | "url2" | "url3"...
> ```

there is id behind them
image

image

**Comment by bluefield-creator** [comment]
I personally disapprove the fact that alyx had to fix your problems @SK-Fast, consdier this a bad grade.

File: src/commands/guild.ts [hunk 61]
```diff
@@ -1,49 +1,59 @@
...
+	}
 
 	const userData = await userUtils.getUserData(data.CreatorID)
 
 	const Embed = new MessageEmbed({
 		title: data.Name,
 		description: data.Description,
 		thumbnail: {
-			url: `${data.Thumbnail}`
+			url: data.thumbnail
```
Comment (bluefield-creator) [inline src/commands/guild.ts]: This does not work, in the response data thumbnail comes as `Thumbnail` and it's not considered a string for the Embed type for some reason.

File: src/utils/stringUtils.ts [hunk 62]
```diff
@@ -1,32 +1,31 @@
-const url_pattern = new RegExp('^(https?:\\/\\/)?'+
-    '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|'+
-    '((\\d{1,3}\\.){3}\\d{1,3}))'+
-    '(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*'+
-    '(\\?[;&a-z\\d%_.~+=-]*)?'+ 
-    '(\\#[-a-z\\d_]*)?$','i');
+const url_pattern = new RegExp(
```
Comment (bluefield-creator) [inline src/utils/stringUtils.ts]: Why is this still here??

File: src/utils/stringUtils.ts [hunk 63]
```diff
@@ -37,47 +36,46 @@ export class stringUtils {
...
-
-        let returnResult: {URLData: URL, searchParams: URLSearchParams, spitted: Array<string>} = {
-            URLData: urlConstruct,
-            searchParams: searchParams_url,
-            spitted: spitted_url
-        }
+	public static urlParse(url: string) {
+		let urlConstruct = new URL(url)
+		let searchParams_url = urlConstruct.searchParams
+		let spitted_url: Array<string> = url.split('/')
```
Comment (bluefield-creator) [inline src/utils/stringUtils.ts]: Wrong type again.

File: src/utils/stringUtils.ts [hunk 64]
```diff
@@ -37,47 +36,46 @@ export class stringUtils {
...
 
-        return returnResult
-    }
+		let returnResult: {URLData: URL; searchParams: URLSearchParams; spitted: Array<string>} = {
+			URLData: urlConstruct,
+			searchParams: searchParams_url,
+			spitted: spitted_url
```
Comment (bluefield-creator) [inline src/utils/stringUtils.ts]: Camel case...

**Review by bluefield-creator — CHANGES_REQUESTED** [review]

**Comment by Troplo** [comment]
How many comments can someone make

File: src/utils/stringUtils.ts [hunk 65]
```diff
@@ -1,81 +1,36 @@
...
 	 * ### Get Numbers From URL
 	 * @param string URL that want to be parsed
 	 * @returns array Numbers that has been found in URL
 	 */
-	public static getNumbersFromURL(string: string) {
-		let result = stringUtils.urlParse(string)
-		let splitted = result.spitted
+	public static getNumbersFromURL(string: string): number[] {
+		const parsed = stringUtils.urlParse(string)
+		const result: number[] = []
```
Comment (bluefield-creator) [inline src/utils/stringUtils.ts:24]: Should be a `let` since we push items.

File: src/utils/stringUtils.ts [hunk 66]
```diff
@@ -16,8 +16,8 @@ export class stringUtils {
 
 	/**
 	 * ### Get Numbers From URL
-	 * @param string URL that want to be parsed
-	 * @returns array Numbers that has been found in URL
+	 * @param {string} URL that want to be parsed
+	 * @returns {number[]} Numbers that has been found in URL
 	 */
 	public static getNumbersFromURL(string: string): number[] {
```
Comment (bluefield-creator) [inline src/utils/stringUtils.ts]: url not string

**Review by bluefield-creator — CHANGES_REQUESTED** [review]

**Comment by Kqpa** [comment]
my guy gave up

**Review by bluefield-creator — APPROVED** [review]
Passes testing, going to review a bit more then merguing.

**Comment by bluefield-creator** [comment]
> Passes testing, going to review a bit more then merguing.

asides recommendation of using more Axios types (which is the reason @Alyxsqrd wanted to use axios to begin with) everything seems ok, I'll pass it on build now, and if it works merguing. 

Will begin setting up CI for this.

**Comment by bluefield-creator** [comment]
There's a small mistake at Eslintrc json, ecmaversion cannot be 13, for latest ecmascript use keyword "latest"

```json
{
    "env": {
        "es2021": true,
        "node": true
   },
    "extends": [
        "standard"
    ],
    "parser": "@typescript-eslint/parser",
    "parserOptions": {
        "ecmaVersion": "latest",
        "sourceType": "module"
   },
    "plugins": [
        "@typescript-eslint"
    ],
    "rules": {
   }
}

```

**Comment by bluefield-creator** [comment]
~~Another thing I noticed, is that someone Imported discord.js to the index.d.ts file, remove the import.~~
NVM it is being used for Icommand type.

**Comment by bluefield-creator** [comment]
image
Noticed that the eslint configuration is slightly off.

**Comment by bluefield-creator** [comment]
https://user-images.githubusercontent.com/48862659/147358352-e50cfc23-df92-4e9c-bbab-db4bc465917e.mp4

eslint configuration goes against prettier configuration.
